### PR TITLE
Amend screens to submit amended client

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**

### CODE REVIEW

File: src/main/java/uk/gov/laa/ccms/caab/advice/ActiveCaseModelAdvice.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  /**
+   * Controller advice method responsible for adding active case to the model of selected
+   * controllers.
+   *
+   * @param model the model view to be updated
+   * @param session the session data
+   */
+  @ModelAttribute
+  public void addActiveCaseToModel(Model model, final HttpSession session) {
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/advice/ActiveCaseModelAdvice.java:45]: final :-)

File: src/main/java/uk/gov/laa/ccms/caab/bean/ClientFlowFormData.java [hunk 2]
```diff
@@ -24,4 +24,18 @@ public ClientFlowFormData(String action) {
     this.action = action;
   }
 
+  /**
+   * Overrides the Default setter, so we can set the vulnerable clients for other child objects.
+   */
+  public void setBasicDetails(ClientFormDataBasicDetails basicDetails) {
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/bean/ClientFlowFormData.java:30]: final

File: src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClient.java [hunk 3]
```diff
@@ -188,8 +188,10 @@ public Mono<ClientStatus> getClientStatus(String transactionId, String loginId,
    * @param userType              Type of the user (e.g., admin, user).
    * @return A Mono wrapping the ClientCreated transaction id.
    */
-  public Mono<ClientCreated> postClient(ClientDetailDetails clientDetails, String loginId,
-                                        String userType) {
+  public Mono<ClientTransactionResponse> postClient(
+      ClientDetailDetails clientDetails,
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClient.java:192]: finals

File: src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClientErrorHandler.java [hunk 4]
```diff
@@ -146,19 +146,33 @@ public Mono<ClientStatus> handleClientStatusError(String transactionId, Throwabl
...
   /**
-   * Handles errors that occur while fetching Client Status.
+   * Handles errors that occur while fetching Client create Status.
    *
    * @param fullName client full name.
    * @param e Exception thrown during operation.
    * @return An empty Mono.
    */
-  public Mono<ClientCreated> handleClientCreatedError(String fullName, Throwable e) {
+  public Mono<ClientTransactionResponse> handleClientCreatedError(String fullName, Throwable e) {
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClientErrorHandler.java:155]: finals

File: src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClientErrorHandler.java [hunk 5]
```diff
@@ -146,19 +146,33 @@ public Mono<ClientStatus> handleClientStatusError(String transactionId, Throwabl
...
   }
 
+  /**
+   * Handles errors that occur while fetching Client update Status.
+   *
+   * @param fullName client full name.
+   * @param e Exception thrown during operation.
+   * @return An empty Mono.
+   */
+  public Mono<ClientTransactionResponse> handleClientUpdatedError(String fullName, Throwable e) {
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClientErrorHandler.java:169]: finals

File: src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/ApplicationTypeSectionController.java [hunk 6]
```diff
@@ -84,15 +79,11 @@ public String applicationSummaryApplicationType(
   @PostMapping("/application/summary/application-type")
   public String delegatedFunction(
       @SessionAttribute(APPLICATION_ID) final String applicationId,
-      @SessionAttribute(ACTIVE_CASE) final ActiveCase activeCase,
       @SessionAttribute(USER_DETAILS) UserDetail user,
       @ModelAttribute(APPLICATION_FORM_DATA) ApplicationFormData applicationFormData,
-      BindingResult bindingResult,
-      Model model) throws ParseException {
+      BindingResult bindingResult) throws ParseException {
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/ApplicationTypeSectionController.java:84]: finals

Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/ApplicationTypeSectionController.java]: done

File: src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClientErrorHandler.java (hunk 5 above)
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClientErrorHandler.java]: done

File: src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClientErrorHandler.java (hunk 4 above)
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClientErrorHandler.java]: done

File: src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClient.java [hunk 7]
```diff
@@ -198,32 +200,36 @@ public Mono<ClientCreated> postClient(ClientDetailDetails clientDetails, String
...
+   * @param clientReferenceNumber The client's reference number.
    * @param clientDetails         The client's details.
    * @param loginId               The login identifier for the user.
    * @param userType              Type of the user (e.g., admin, user).
    * @return A Mono wrapping the ClientCreated transaction id.
    */
-  public Mono<ClientCreated> putClient(ClientDetailDetails clientDetails, String loginId,
-                                        String userType) {
+  public Mono<ClientTransactionResponse> putClient(
+      String clientReferenceNumber,
```
[the remark this answers is not in the document]
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClient.java]: done

File: src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClient.java (hunk 3 above)
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/client/SoaApiClient.java]: done

File: src/main/java/uk/gov/laa/ccms/caab/bean/ClientFlowFormData.java (hunk 2 above)
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/bean/ClientFlowFormData.java]: done

File: src/main/java/uk/gov/laa/ccms/caab/advice/ActiveCaseModelAdvice.java (hunk 1 above)
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/advice/ActiveCaseModelAdvice.java]: done

File: src/main/java/uk/gov/laa/ccms/caab/controller/submission/ClientSubmissionsInProgressController.java [hunk 8]
```diff
@@ -57,15 +59,44 @@ public String submissionInProgress(
...
+    return viewIncludingPollCount(session, SUBMISSION_UPDATE_CLIENT);
+  }
+
+  private String viewIncludingPollCount(HttpSession session, String submissionType) {
     int submissionPollCount = 0;
     if (session.getAttribute("submissionPollCount") != null) {
       submissionPollCount = (int) session.getAttribute("submissionPollCount");
       if (submissionPollCount >= 6) {
-        return "redirect:/submissions/client-create/failed";
+        return String.format("redirect:/submissions/%s/failed", submissionType);
```
Comment (porritta) [inline src/main/java/uk/gov/laa/ccms/caab/controller/submission/ClientSubmissionsInProgressController.java:99]: hardcoded max poll count to be externalised, as discussed

File: src/test/java/uk/gov/laa/ccms/caab/util/SoaModelUtils.java [hunk 9]
```diff
@@ -555,7 +554,7 @@ public static ClientDetail buildClientDetail() {
                 .vexatiousLitigant(Boolean.FALSE)
                 .vulnerableClient(Boolean.FALSE))
             .specialConsiderations("special"))
-        .recordHistory(new ClientDetailRecordHistory());
+        .recordHistory(new uk.gov.laa.ccms.soa.gateway.model.RecordHistory());
```
Comment (porritta) [inline src/test/java/uk/gov/laa/ccms/caab/util/SoaModelUtils.java:557]: This class is imported, so don't need to fully qualify

**Review by porritta — COMMENTED** [review]
review complete

File: src/main/java/uk/gov/laa/ccms/caab/controller/submission/ClientSubmissionsInProgressController.java (hunk 8 above)
Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/controller/submission/ClientSubmissionsInProgressController.java:107]: done

File: src/test/java/uk/gov/laa/ccms/caab/util/SoaModelUtils.java (hunk 9 above)
Comment (PhilDigitalJustice) [reply src/test/java/uk/gov/laa/ccms/caab/util/SoaModelUtils.java]: done

**Review by porritta — APPROVED** [review]
LGTM

**Pull request merged**